### PR TITLE
IOC implementation

### PR DESCRIPTION
**Pull request by Valgri**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1,10 +1,8 @@
-import { Users } from './services/users';
-import { Logger } from './services/logger';
-
+import { createIoCContainer } from './ioc';
 import type { User, ApiConfig } from './types';
 
 const renderUsers = async (config: ApiConfig) => {
-  const usersService = new Users(config);
+  const usersService = createIoCContainer(config).resolve('Users');
```
Comment (Nazar-bnm) [inline src/index.ts]: you shouldn't create a new ioc container each time, please create it once and reuse

File: src/ioc/index.ts [hunk 2]
```diff
@@ -1,12 +1,18 @@
...
-export const createIoCContainer = () =>  {
+export const createIoCContainer = (config: ApiConfig = { path: '', resources: {} }) =>  {
   const ioc = new IoCContainer();
-  // you can register some resources right now below...
+
+  ioc.register('Logger', new Logger());
+
+  ioc.register('HTTP',  new HTTP(config));
+
+  ioc.register('Users', new Users(config));
```
Comment (Nazar-bnm) [inline src/ioc/index.ts]: you shouldn't pass the config by yourself it should be injected automatically. This course about dependency injection 🙂
You can find an example of how to automate dependency injection here - https://www.npmjs.com/package/ioc-lite

**Review by Nazar-bnm — CHANGES_REQUESTED** [review]

File: src/index.ts (hunk 1 above)
Comment (Valgri) [reply src/index.ts]: thanks! Please check 2nd commit, where I have made changes.

File: src/ioc/index.ts (hunk 2 above)
Comment (Valgri) [reply src/ioc/index.ts]: Got it! Please check 2nd commit, where I have made changes.

**Review by Nazar-bnm — APPROVED** [review]

**Pull request merged**